bactpipe.nf

```diff
@@ -1,363 +1,107 @@
 log.info "".center(60, "=")
 
 params.help = false
-def printHelp() {
-    log.info """
-    Example usage:
-    nextflow run ctmrbio/BACTpipe --mashscreen_database path/to/refseq.genomes.k21s1000.msh --reads '*_R{1,2}.fastq.gz'
 
-    Mandatory arguments:
-      --reads                Path to input data (must be surrounded with single quotes).
-      --mashscreen_database  Path to mash screen database (can be omitted if set in config file).
+//================================================================================
+// Include modules and (soft) override module-level parameters
+//================================================================================
 
-    Output options:
-      --output_dir           Output directory, where results will be saved (default: ${params.output_dir}).
 
-    Refer to the online manual for more information on available options:
-               https://bactpipe.readthedocs.io
-    """
-}
-
-def printSettings() {
-    log.info "Running with the following settings:".center(60)
-    for (option in params) {
-        if (option.key in ['cluster-options', 'help']) {
-            continue
-        }
-        log.info "${option.key}: ".padLeft(30) + "${option.value}"
-    }
-    log.info "".center(60, "=")
-}
+include { ASSEMBLY_STATS } from "./modules/assembly_stats/assembly_stats.nf"
+include { FASTP } from "./modules/fastp/fastp.nf"
+include { MULTIQC } from "./modules/multiqc/multiqc.nf"
+include { PROKKA } from "./modules/prokka/prokka.nf"
+include { SCREEN_FOR_CONTAMINANTS } from "./modules/screen_for_contaminants/screen_for_contaminants.nf"
+include { SHOVILL } from "./modules/shovill/shovill.nf"
+include { printHelp; printSettings } from "./modules/utils/utils.nf"
 
 
-try {
-    if ( ! nextflow.version.matches(">= $nf_required_version") ){
-        throw GroovyException('Nextflow version too old')
-    }
-} catch (all) {
-    log.error "\n" +
-              "".center(60, "=") + "\n" +
-              "BACTpipe requires Nextflow version $nf_required_version!".center(60) + "\n" +
-              "You are running version $workflow.nextflow.version.".center(60) + "\n" +
-              "Please run `nextflow self-update` to update Nextflow.".center(60) + "\n" +
-              "".center(60, "=") + "\n"
-    exit(1)
-}
+//================================================================================
+// Pre-flight checks and info
+//================================================================================
 
 if (workflow['profile'] in params.profiles_that_require_project) {
     if (!params.project) {
         log.error "BACTpipe requires that you set the 'project' parameter when running the ${workflow['profile']} profile.\n".center(60) +
-                  "Specify --project <project_name> on the command line, or set it in a custom configuration file.".center(60)
+                "Specify --project <project_name> on the command line, or tuple it in a custom configuration file.".center(60)
```

Comment:
:D 
The word "set" in the original version of this line did not refer the the data structure. Interesting to see the word "tuple" used as a verb to configure something as a tuple in the custom config file. Is this common? I've never encountered it used like this before.